src/bokeh/io/showing.py

```diff
@@ -49,16 +54,21 @@
 # General API
 #-----------------------------------------------------------------------------
 
-def show(obj: UIElement | Application | ModifyDoc, browser: str | None = None, new: BrowserTarget = "tab",
-         notebook_handle: bool = False, notebook_url: str | ProxyUrlFunc = DEFAULT_JUPYTER_URL,
-         **kwargs: Any) -> CommsHandle | None:
+def show(
+    obj: UIElement | Sequence[UIElement] | Application | ModifyDoc,
```

Comment:
From typing perspective overrides and type unions are equivalent. With overrides there would be a lot of repetition of other arguments. It would make sense to use overrides if other arguments depend on specific cases, which may be the case with notebook arguments and the return type. I will leave that for typing specific PR.